test/e2e/performanceprofile/functests/10_mixedcpus/mixedcpus.go

```diff
@@ -0,0 +1,160 @@
+)
+
+const (
+	crioRuntimeConfigFile      = "/etc/crio/crio.conf.d/99-runtimes.conf"
+	kubeletMixedCPUsConfigFile = "/etc/kubernetes/openshift-workload-mixed-cpus"
+)
+
+var _ = Describe("Mixedcpus", Ordered, func() {
+	ctx := context.Background()
+	BeforeAll(func() {
+		var err error
+		profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred())
+
+		if profileutil.IsMixedCPUsEnabled(profile) {
+			testlog.Infof("mixed cpus already enabled for profile %q", profile.Name)
+		}
+		if !profileutil.IsMixedCPUsEnabled(profile) {
+			testlog.Infof("enable mixed cpus for profile %q", profile.Name)
+			teardown := setup(ctx, profile)
+			DeferCleanup(teardown, ctx)
+		}
+	})
+
+	Context("configuration files integrity", func() {
+		var profile *performancev2.PerformanceProfile
+		BeforeEach(func() {
+			var err error
+			profile, err = profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+			Expect(err).ToNot(HaveOccurred())
+
+		})
+
+		It("should deploy kubelet configuration file", func() {
+			workers, err := nodes.GetByLabels(testutils.NodeSelectorLabels)
+			Expect(err).ToNot(HaveOccurred())
+			// test arbitrary one should be good enough
+			worker := &workers[0]
+			cmd := isFileExistCmd(kubeletMixedCPUsConfigFile)
+			found, err := nodes.ExecCommandOnNode(cmd, worker)
+			Expect(err).ToNot(HaveOccurred(), "failed to execute command on node; cmd=%q node=%q", cmd, worker)
+			Expect(found).To(Equal("true"), "file not found; file=%q", kubeletMixedCPUsConfigFile)
+		})
+
+		It("should add Kubelet systemReservedCPUs the shared cpuset", func() {
+			name := components.GetComponentName(profile.Name, components.ComponentNamePrefix)
+			key := client.ObjectKey{Name: name}
+			kc := &machineconfigv1.KubeletConfig{}
+			Expect(testclient.Client.Get(ctx, key, kc)).ToNot(HaveOccurred())
+			k8sKc := &kubeletconfig.KubeletConfiguration{}
+			Expect(json.Unmarshal(kc.Spec.KubeletConfig.Raw, k8sKc)).ToNot(HaveOccurred())
+			// don't need to check the error, the values were already validated.
+			reserved, _ := cpuset.Parse(string(*profile.Spec.CPU.Reserved))
+			shared, _ := cpuset.Parse(string(*profile.Spec.CPU.Shared))
+			reservedSystemCpus, _ := cpuset.Parse(k8sKc.ReservedSystemCPUs)
+			Expect(reservedSystemCpus.Equals(reserved.Union(shared))).To(BeTrue(), "reservedSystemCPUs should contain the shared cpus; reservedSystemCPUs=%q reserved=%q shared=%q",
+				reservedSystemCpus.String(), reserved.String(), shared.String())
+		})
+
+		It("should update runtime configuration with the given shared cpuset", func() {
+			workers, err := nodes.GetByLabels(testutils.NodeSelectorLabels)
+			Expect(err).ToNot(HaveOccurred())
+			// test arbitrary one should be good enough
+			worker := &workers[0]
+			cmd := []string{
+				"chroot",
+				"/rootfs",
+				"/bin/bash",
+				"-c",
+				fmt.Sprintf("/bin/grep -E 'shared_cpuset.*' %s | awk -F '\"' '{ print $2 }'", crioRuntimeConfigFile),
```

Comment:
minor (for a future improvement):
1. if you use the absolute path for grep (good), why not for awk?
2. you can just call awk, no need for a pipe

Reply:
TIL